controller/controlplane/controller_watch.go

```diff
@@ -288,3 +288,81 @@ func (r *Reconciler) listControlPlanesForReferenceGrants(
 
 	return recs
 }
+
+func (r *Reconciler) listControlPlanesForRoles(
+	ctx context.Context,
+	obj client.Object,
+) []reconcile.Request {
+	role, ok := obj.(*rbacv1.Role)
+	if !ok {
+		ctrllog.FromContext(ctx).Error(
+			operatorerrors.ErrUnexpectedObject,
+			"failed to map Role on ControlPlane",
+			"expected", "Role", "found", reflect.TypeOf(obj),
+		)
+		return nil
+	}
+
+	nn, ok := isManagedByControlPlane(role)
+	if !ok {
+		return nil
+	}
+
+	return []reconcile.Request{
+		{
+			NamespacedName: nn,
+		},
+	}
+}
+
+func (r *Reconciler) listControlPlanesForRoleBindings(
+	ctx context.Context,
+	obj client.Object,
+) []reconcile.Request {
+	roleBinding, ok := obj.(*rbacv1.RoleBinding)
+	if !ok {
+		ctrllog.FromContext(ctx).Error(
+			operatorerrors.ErrUnexpectedObject,
+			"failed to map RoleBinding on ControlPlane",
+			"expected", "RoleBinding", "found", reflect.TypeOf(obj),
+		)
+		return nil
+	}
+
+	nn, ok := isManagedByControlPlane(roleBinding)
+	if !ok {
+		return nil
+	}
+
+	return []reconcile.Request{
+		{
+			NamespacedName: nn,
+		},
+	}
+}
```

Comment:
```suggestion
func listControlPlanesFor[T *rbacv1.Role | *rbacv1.RoleBinding](
	ctx context.Context,
	obj client.Object,
) []reconcile.Request {
	if _, ok := obj.(T); !ok {
		ctrllog.FromContext(ctx).Error(
			operatorerrors.ErrUnexpectedObject,
			fmt.Sprintf("failed to map %T on ControlPlane", T(nil)),
			"expected", "RoleBinding", "found", reflect.TypeOf(obj),
		)
		return nil
	}

	nn, ok := isManagedByControlPlane(obj)
	if !ok {
		return nil
	}

	return []reconcile.Request{
		{
			NamespacedName: nn,
		},
	}
}
```

Instead of having two specific method helpers, have one generic function helper. WDYT? 
On the caller side, it'll look like that 

```go
Watches(
			&rbacv1.RoleBinding{},
			handler.EnqueueRequestsFromMapFunc(listControlPlanesFor[*rbacv1.RoleBinding]))
```

Reply:
👍 

Added that in 3f20b733182c2fca371d9c7beffa59d182d75704 (needed to change your snippet ever so slightly to make it compile).